src/lib.rs

```
@@ -190,6 +212,178 @@ impl FromStr for Cookie {
+                    }
+                }
+            }
+
+            deserializer.deserialize(CookieFieldVisitor)
+        }
+    }
+
+    pub struct CookieDeserializer;
+
+    impl serde::de::Visitor for CookieDeserializer {
+         type Value = Cookie;
+
+         fn visit_map<V>(&mut self, mut v: V) -> Result<Cookie, V::Error>
+             where V: serde::de::MapVisitor
+         {
+             let mut name = None;
+             let mut value = None;
+             let mut expires: Option<String> = None;
+             let mut max_age = None;
+             let mut domain = None;
+             let mut path = None;
+             let mut secure = None;
+             let mut http = None;
+
+             while let Some(key) = try!(v.visit_key()) {
+                 match key {
+                     CookieField::Name => { name = Some(try!(v.visit_value())); }
+                     CookieField::Value => { value = Some(try!(v.visit_value())); }
+                     CookieField::Expires => { expires = Some(try!(v.visit_value())); },
+                     CookieField::MaxAge => { max_age = Some(try!(v.visit_value())); },
+                     CookieField::Domain => { domain = Some(try!(v.visit_value())); },
+                     CookieField::Path => { path = Some(try!(v.visit_value())); },
+                     CookieField::Secure => { secure = Some(try!(v.visit_value())); },
+                     CookieField::HttpOnly => { http = Some(try!(v.visit_value())); },
+                 }
+             }
+
+             let name = match name {
+                 Some(name) => name,
+                 None => try!(v.missing_field("name"))
+             };
+
+             let value = match value {
+                 Some(value) => value,
+                 None => try!(v.missing_field("value"))
+             };
+
+             let expires = expires.map(|expires| {
+                 time::strptime(&*expires, "%a, %d %b %Y %H:%M:%S %Z").unwrap()
```

Ah yeah this'll always succeed for data created from serialization, but it may be possible that something like this could change over time or perhaps the data is user generated. In that sense it should propagate the error.

This may become moot though if serialization just goes through a string